### PR TITLE
Taxonomies and Taxon API

### PR DESCRIPTION
**Pull request by blackode**
1. Imported taxonomy seed data files
2. Created relevant **views**  & **controllers** for `Taxonomy` and `Taxon`
3. Created a route resource `taxonomies [only: index]` 

## Any further comments?
The `snitch_core` app from `develop` lack with some functions like `Taxonomy.get_all_taxonomy()` used to fetch the list from `super_sonic` branch which is not availble in `develop` branch. 
The `snitch_core` suppose to be merged with `super_sonic`. IMO. 

------

## Checklist

- [x] I have read [CONTRIBUTING.md][contributing].
- [x] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [x] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

------
Dear Gatekeeper,

Please make sure that the commits that will be merged or rebased into the base
branch are [formatted according to our standards][commit-style].

> The only additional requirement is that the the PR number must appear in the
> commit title in brackets: `(#XXX)`

[commit-style]: https://github.com/aviabird/snitch/blob/develop/CONTRIBUTING.md#styling

### CODE REVIEW

File: apps/snitch_api/lib/snitch_api_web/controllers/taxon_controller.ex [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def show(conn, %{"id" => id}) do
+    taxon =
+      Repo.get!(Taxon, id)
+      |> Repo.preload([:parent, :taxonomy])
+
+    render(
+      conn,
+      "index.json-api",
+      data: taxon,
+      opts: [include: "parent, taxonomy"]
```
Comment (pkrawat1) [inline apps/snitch_api/lib/snitch_api_web/controllers/taxon_controller.ex]: should be without space like `Example: include: "parent,taxonomy"`

File: apps/snitch_api/lib/snitch_api_web/controllers/taxon_controller.ex [hunk 2]
```diff
@@ -0,0 +1,32 @@
+defmodule SnitchApiWeb.TaxonController do
+  use SnitchApiWeb, :controller
+
+  alias Snitch.Repo
+  alias Snitch.Data.Schema.Taxon
+
+  def index(conn, _params) do
+    taxons =
+      Repo.all(Taxon)
```
Comment (pkrawat1) [inline apps/snitch_api/lib/snitch_api_web/controllers/taxon_controller.ex]: Why using `Repo` here.

```elixir
alias Snitch.Data.Model.TaxonModel
TaxonModel.get_all |> Repo.preload([:parent, :taxonomy])
```

Comment (blackode) [reply apps/snitch_api/lib/snitch_api_web/controllers/taxon_controller.ex]: I saw that. Initially, I tried like you suggested. But, I did not find the model in the `develop` branch for `snitch_core`. But, it is present in the `super_sonic` branch. Need to merge those files. IMO :idea:
https://github.com/aviabird/snitch/tree/develop/apps/snitch_core/lib/core/data/model

Comment (pkrawat1) [reply apps/snitch_api/lib/snitch_api_web/controllers/taxon_controller.ex]: Copy that model code as it is in your branch. Use `super_sonic` as reference. And see if you need to copy any code from there.

File: apps/snitch_api/lib/snitch_api_web/router.ex [hunk 3]
```diff
@@ -10,5 +10,7 @@ defmodule SnitchApiWeb.Router do
     pipe_through(:api)
 
     resources("/orders", OrderController, only: [:index])
+    resources("/taxonomies", TaxonomyController, only: [:index, :show])
+    resources("/taxons", TaxonController, only: [:index, :show])
```
Comment (oyeb) [inline apps/snitch_api/lib/snitch_api_web/router.ex]: I think `taxons` resource must be nested in `taxonomies`. What do you say @SagarKarwande ?

Comment (SagarKarwande) [reply apps/snitch_api/lib/snitch_api_web/router.ex]: Confirmed this from @pkrawat1. Right now it is ok

Comment (SagarKarwande) [reply apps/snitch_api/lib/snitch_api_web/router.ex]: Taxon will be aggregated as per the hierarchy on frontend.

Comment (pkrawat1) [reply apps/snitch_api/lib/snitch_api_web/router.ex]: Looks fine

**Pull request merged**